babel/numbers.py

```diff
@@ -1001,6 +1001,7 @@ def __init__(self, message: str, suggestions: list[str] | None = None) -> None:
 def parse_number(
     string: str,
     locale: Locale | str | None = LC_NUMERIC,
+    strict: bool = False,
```

Comment:
Please add the new argument into the docstring. `strict` mode should probably be the default here, and the new behavior (which allows more loose strings to be parsed) should be opt-in.